src/org/labkey/test/util/TestDataGenerator.java

```diff
@@ -328,6 +330,31 @@ public static String randomString(int size, @Nullable String exclusion, @Nullabl
         return val.toString();
     }
 
+    public static String randomDomainName()
+    {
+        return randomDomainName(10);
+    }
+
+    public static String randomInvalidDomainName(int size)
+    {
+        return randomString(size, null, ILLEGAL_DOMAIN_NAME_CHARSET);
+    }
+
+    public static String randomDomainName(int size)
+    {
+        String domainName = "";
+        do
+        {
+            String prefix = randomString(1, null, ALPHANUMERIC_STRING); // domain needs to start with alphanumeric char
+            final String charset = ALPHANUMERIC_STRING + DOMAIN_SPECIAL_STRING;
+            domainName = prefix + randomString(size - 1, null, charset);
+            domainName = domainName.trim();
```

Comment:
How would the name have any spaces to be trimmed?

Reply:
They might have trailing space due to DOMAIN_SPECIAL_STRING contains whitespace